File: playground/pingpong/PMachine.cs
using System;
using System.Collections.Generic;

class PMachine {
    public int state;
    public List<int> events = new List<int>();
    public List<object[]> payloads = new List<object[]>();

    public void sendMsg(PMachine other, int e, object[] payload) {
        Console.WriteLine(this.ToString() + " send event " + e.ToString() + " to " + other.ToString());
        other.events.Add(e);
        other.payloads.Add(payload);
    }
}

File: playground/pingpong_java/PMachine.java
import java.util.*;

class PMachine {
    public int state;
    public List<Integer> events = new ArrayList<>();
    public List<Object> payloads = new ArrayList<>();

    public void sendMsg(PMachine other, int e, Object payload) {
        System.out.println(this.toString() + " send event " + Integer.toString(e) + " to " + other.toString());
        other.events.add(e);
        other.payloads.add(payload);
    }
}